router/router.go

```diff
@@ -631,20 +569,11 @@ func (worker *workerT) processDestinationJobs() {
 				//			router_delivery_exceeded_timeout -> goes to zero
 				ch := worker.trackStuckDelivery()
 
-				resultSetID := destinationJob.JobMetadataArray[0].ResultSetID
-
-				if worker.localResultSet.id < resultSetID {
-					resultSet := worker.rt.getResultSet(resultSetID)
-					worker.localResultSet.id = resultSetID
-					worker.localResultSet.resultSetBeginTime = time.Now()
-					worker.localResultSet.timeAllotted = resultSet.timeAllotted
-				}
-
 				// Assuming twice the overhead - defensive: 30% was just fine though
 				// In fact, the timeout should be more than the maximum latency allowed by these workers.
 				// Assuming 10s maximum latency
-				elapsed := time.Since(worker.localResultSet.resultSetBeginTime)
-				threshold := time.Duration(2.0 * math.Max(float64(worker.localResultSet.timeAllotted), float64(10*time.Second)))
+				elapsed := time.Since(worker.latestAssignedTime)
+				threshold := worker.rt.routerTimeout
 				if elapsed > threshold {
 					respStatusCode = types.RouterTimedOutStatusCode
```

Comment:
I can't find a test that is supposed to test this bit. Do we have one?

Reply:
There aren't any right now... Will try to add a test case for this

Reply:
Added a test for this